python/examples/token_swap/token_swap.py

```diff
@@ -0,0 +1,370 @@
+
+    if trustline_approver_receiving_token[sf_balance] < zero_amount:
+        trustline_approver_receiving_token[sf_balance] = trustline_approver_receiving_token[sf_balance] - tx_proposer_send_token
+    else:
+        trustline_approver_receiving_token[sf_balance] = trustline_approver_receiving_token[sf_balance] + tx_proposer_send_token
+
+    print(f"-> after {trustline_approver_receiving_token[sf_balance]}")
+
+    print("\n\n")
+
+    print("trustline_proposer_sending_token")
+    print("--------------------------------")
+    print(f"-> before {trustline_proposer_sending_token[sf_balance]}")
+   
+    if trustline_proposer_sending_token[sf_balance] < zero_amount:
+        trustline_proposer_sending_token[sf_balance] = trustline_proposer_sending_token[sf_balance] + tx_proposer_send_token
+    else:
+        trustline_proposer_sending_token[sf_balance] = trustline_proposer_sending_token[sf_balance] - tx_proposer_send_token
+
+    print(f"-> after {trustline_proposer_sending_token[sf_balance]}")
+
+    print("\n\n")
+
+    print("trustline_proposer_receiving_token")
+    print("----------------------------------")
+    print(f"-> before {trustline_proposer_receiving_token[sf_balance]}")
+    
+    if trustline_proposer_receiving_token[sf_balance] < zero_amount:
+        trustline_proposer_receiving_token[sf_balance] = trustline_proposer_receiving_token[sf_balance] - tx_approver_send_token
+    else:
+        trustline_proposer_receiving_token[sf_balance] = trustline_proposer_receiving_token[sf_balance] + tx_approver_send_token
+    
+    print(f"-> after {trustline_proposer_receiving_token[sf_balance]}")
+
+    print("\n\n")
+
+    ctx.view().update(trustline_approver_sending_token)
+    ctx.view().update(trustline_approver_receiving_token)
+    ctx.view().update(trustline_proposer_sending_token)
+    ctx.view().update(trustline_proposer_receiving_token)
+
+    page = slep[sf_owner_node]
+    
+    # Delete token swap object on originating account
+    ctx.view().dir_remove(owner_dir_keylet(tx_account_other), page, token_swap_keylet, True)
+    ctx.view().erase(slep)
+
+    # Adjust reserve count
+    adjust_owner_count(ctx.view(), proposer_account, -1, ctx.journal)
+
+    return tesSUCCESS
+    
+transactors = [
+    Transactor(
+        name="TokenSwapPropose",
+        tx_type=53,
+        tx_format=[
+            (sf_amount, soeREQUIRED),
+            (sf_amount_other, soeREQUIRED),
+            (sf_account_other, soeREQUIRED),
+            (sf_expiration, soeREQUIRED),
+        ],
+        consequences_factory_type=ConsequencesFactoryType.Normal,
+        preflight=preflight_propose,
+        do_apply=do_apply_propose,
+    ),
+    Transactor(
+        name="TokenSwapAccept",
+        tx_type=54,
+        tx_format=[
+            (sf_amount, soeREQUIRED),
+            (sf_amount_other, soeREQUIRED),
+            (sf_account_other, soeREQUIRED),
+            (sf_token_swap_id, soeREQUIRED),
```

Comment:
You should only need the token swap ID in this transaction, since the other details are all in the token swap object.